wal/wal.go

```diff
@@ -196,7 +196,7 @@ func (w *FileWAL) Run(ctx context.Context) {
 			err := w.log.WriteBatch(walBatch)
 			if err != nil {
 				w.metrics.failedLogs.Add(float64(len(batch)))
-				level.Error(w.logger).Log("msg", "failed to write WAL batch", "err", err)
+				level.Error(w.logger).Log("msg", "failed to write WAL batch", "err", err, "batch", walBatch)
```

Comment:
What would an example log line look like here? Isn't `walBatch` a struct with only unexported fields, how will this log line help us?

Reply:
Yes, it's pretty opaque given the private fields, but we can at least determine the indexes/reconstruct. Something like this is printed out (and we can tell that the index is '2'):
```
batch &{[{2 522}] [10 135 4 10 132 4 10 11 115 116 97 99 107 116 114 97 99 101 115 18 244 3 80 65 82 49 21 2 25 220 72 5 112 97 114 99 97 21 24 0 21 4 21 128 1 21 0 24 8 100 117 114 97 116 105 111 110 37 36 76 172 19 64 17 0 0 0 21 12 37 2 24 15 108 97 98 101 108 115 46 105 110 115 116 97 110 99 101 37 0 76 28 0 0 0 21 12 37 2 24 10 108 97 98 101 108 115 46 106 111 98 37 0 76 28 0 0 0 21 12 37 0 24 4 110 97 109 101 37 0 76 28 0 0 0 21 4 21 128 1 21 0 24 6 112 101 114 105 111 100 37 36 76 172 19 64 17 0 0 0 21 12 37 0 24 11 112 101 114 105 111 100 95 116 121 112 101 37 0 76 28 0 0 0 21 12 37 0 24 11 112 101 114 105 111 100 95 117 110 105 116 37 0 76 28 0 0 0 21 12 37 0 24 11 115 97 109 112 108 101 95 116 121 112 101 37 0 76 28 0 0 0 21 12 37 0 24 11 115 97 109 112 108 101 95 117 110 105 116 37 0 76 28 0 0 0 21 12 37 0 24 10 115 116 97 99 107 116 114 97 99 101 37 0 76 28 0 0 0 21 4 21 128 1 21 0 24 9 116 105 109 101 115 116 97 109 112 37 36 76 172 19 64 17 0 0 0 21 4 21 128 1 21 0 24 5 118 97 108 117 101 37 36 76 172 19 64 17 0 0 0 22 0 25 12 25 28 24 15 100 121 110 97 109 105 99 95 99 111 108 117 109 110 115 24 51 108 97 98 101 108 115 58 105 110 115 116 97 110 99 101 44 106 111 98 59 112 112 114 111 102 95 108 97 98 101 108 115 58 59 112 112 114 111 102 95 110 117 109 95 108 97 98 101 108 115 58 0 24 65 103 105 116 104 117 98 46 99 111 109 47 115 101 103 109 101 110 116 105 111 47 112 97 114 113 117 101 116 45 103 111 32 118 101 114 115 105 111 110 32 48 46 48 46 48 40 98 117 105 108 100 32 57 53 50 98 49 54 49 51 97 49 57 49 41 25 204 28 0 0 28 0 0 28 0 0 28 0 0 28 0 0 28 0 0 28 0 0 28 0 0 28 0 0 28 0 0 28 0 0 28 0 0 0 232 1 0 0 80 65 82 49]}
```

Reply:
got it!